pkg/auth/providerrefresh/refresher.go

```diff
@@ -313,9 +354,18 @@ func (r *refresher) refreshAttributes(attribs *v3.UserAttribute) (*v3.UserAttrib
 
 	// user has been deactivated, disable their tokens
 	for _, token := range derivedTokenList {
-		token = token.DeepCopy()
-		token.Enabled = pointer.Bool(false)
-		_, err := r.tokenMGR.UpdateToken(token)
+		// Update is type-dependent
+		var err error
+		switch token.(type) {
+		case *v3.Token:
+			v3Token := token.(*v3.Token).DeepCopy()
+			v3Token.Enabled = pointer.Bool(false)
+			_, err = r.tokenMGR.UpdateToken(v3Token)
+		case *ext.Token:
+			err = r.extTokenStore.Disable(token.GetName())
+		default:
+			err = fmt.Errorf("unable to update token of unknown type %T", token)
+		}
```

Comment:
Nit: it is possible to avoid the casting, assigning the asserted type:

```suggestion
		switch t := token.(type) {
		case *v3.Token:
			v3Token := t.DeepCopy()
			v3Token.Enabled = pointer.Bool(false)
			_, err = r.tokenMGR.UpdateToken(v3Token)
		case *ext.Token:
			err = r.extTokenStore.Disable(t.GetName())
		default:
			err = fmt.Errorf("unable to update token of unknown type %T", token)
		}
```